framework/assets/yii.js

```diff
@@ -273,7 +273,7 @@ window.yii = (function ($) {
                 return {};
             }
 
-            var pairs = url.substring(pos + 1).split('#')[0].split('&'),
+            var pairs = url.substring(pos + 1).split('#')[0].split('&').filter(String),
```

Comment:
First of all, thanks for the made improvements. Here are some remarks however:

1) In `pairs` array we already have strings only so I feel that repeated string conversion is redundant. Something more explicit like:

```js
var pairs = url.substring(pos + 1).split('#')[0].split('&').filter(function (value) {
    return value !== '';
})
```

would be better IMO.

2) Used [Array.prototype.filter](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method is OK, but:

>`filter()` was added to the ECMA-262 standard in the 5th edition; as such it may not be present in all implementations of the standard.

To be more specific, it's not supported in Internet Explorer 8 and I'm not sure that we can just refuse supporting it (there is a small chance that this requirement takes place to be in someone's project based on Yii2). I don't think that polyfill will be a good option neither, because its usage currently will be limited by this code only and it seems like an overhead. Regular loop can solve the problem, or maybe consider an option to try not include such values initially to avoid additional filter loop.

@samdark Please remind, do we support Internet Explorer 8?